doc/specs/#1256 - Tab tearoff/spec.md

```diff
@@ -0,0 +1,319 @@
+
+```
+- conhost.exe (in PTY mode) - Hosted to A
+  |- pwsh.exe
+
+- wt.exe (Terminal Instance A)
+```
+
+The `conhost.exe` was started in response to a `pwsh.exe` being started with no host. It then put itself into PTY mode and launched into a connection of `wt.exe` instance A.
+
+**ALTERNATIVELY**
+
+```
+- conhost.exe - idling
+  
+- wt.exe (Terminal Instance A)
+  |- conhost.exe (in PTY mode)
+     |- pwsh.exe
+```
+
+The `conhost.exe` at the top was launched in response to `pwsh.exe` being started with no host. It identified 
+
+### Performance, Power, and Efficiency
+
+This is obviously less efficient than not doing it as we have to stand up servers and protocols and handlers for shuffling things about.
+
+But as long as we're creating threads and services that sleep most of the time and are only awakened on some kernel/system event, we shouldn't be wasting too much in terms of power and background resources.
+
+Additionally, `wt.exe` is worse than `conhost.exe` alone in all efficiency categories simply because it not only requires more resources to display in a "pretty" manner, but it also requires a `conhost.exe` under it in PTY mode to adapt the API calls. This is generally acceptable for end users who care more about the experience than the total performance. 
+
+It is, however, not likely to be much if any worse than just choosing to use `wt.exe` anyway over `conhost.exe`.
+
+## Potential Issues
+
+I've listed most of the issues above in their individual sections. The primary highlights are:
+1. Process tree layout - The processes in hierarchy may not make sense to someone inspecting them either visually with a tool or programatically
+1. Process and kernel object lifetime - Applications may be counting on a specific process or object lifetime in regards to their hosting window and we might be tampering with that in how we apply job objects or shuffle around ownership to make tabs happen
+1. Default launch expectations - It is possible that test utilities or automation are counting on `conhost.exe` being the host application or that they're not ready to tolerate the potential for other applications to start. I think the interactive/non-interactive check mitigates this, but we'd have to remain concerned here.
+1. `AttachConsole` and `DetachConsole` and `AllocConsole` - I don't have the slightest idea what happens for these APIs. We would have to explore. `AttachConsole` has restrictions based on the process hierarchy. It would likely behave in interesting ways with the strange parenting order and might be a driver to why we would have to adjust the parenting of the processes (or change the API under the hood). `DetachConsole` might create an issue where a tab disappears out of the terminal and the job object causes everything to die. `AttachConsole` wouldn't necessarily be guaranteed to go back into the same `wt.exe` or a `wt.exe` at all. 
+
+## Future considerations
+

```

Comment:
Another question, re #1080. How would we handle dragging a tab that doesn't have a terminal in it out? We'd need to somehow move the entire XAML tree of the tab, and all it's panes to the new process. That might be possible with winrt (COM) types.

Reply:
I think that is a component UI problem as well.